compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2234,12 +2241,69 @@ public void visit(BLangMatch matchNode) {
         matchNode.exprTypes = exprTypes;
     }
 
+    @Override
+    public void visit(BLangMatchStatement matchStatement) {
+        typeChecker.checkExpr(matchStatement.expr, env, symTable.noType);
+        matchStatement.matchClauses.forEach(matchClause -> matchClause.accept(this));
```

Comment:
Can we call analyze instead of directly calling accept?

Reply:
changed.